simulate/bed.go

```diff
@@ -1,43 +1,71 @@
 package simulate
 
 import (
+	"log"
 	"github.com/vertgenlab/gonomics/bed"
 	"github.com/vertgenlab/gonomics/numbers"
 )
 
+// CountWindows counts the total viable windows of length regionLength in the sequence searchSpace
+func CountWindows(searchSpace []bed.Bed, regionLength int) int {
+	var length, totalWindows int
+	for i := range searchSpace {
+		length = searchSpace[i].ChromEnd - searchSpace[i].ChromStart
+
+		if length >= regionLength {
+			totalWindows += length - regionLength + 1
+		}
+	}
+
+	return totalWindows
+}
+
+// GenerateBedRegion searches the regions of searchSpace (a noGap.bed input file, as a parsed struct) and randomly selects a continuous region of length regionLength 
+func GenerateBedRegion(searchSpace []bed.Bed, randPos int, regionLength int) (bed.Bed, bool) {
+	var chromWindows int
+	var length int
+	// iterating through each ungapped window
+	for j := range searchSpace {
+		length = searchSpace[j].ChromEnd - searchSpace[j].ChromStart
+
+		// check that the window can fit a sequence of that length e.g. want 10 bp, but window is 5bp
+		chromWindows = length - regionLength + 1
+
+		if chromWindows < 1 {
+			continue
+		}
+
+		// Decrement randomly generated overall position until it fits within the region
+		if randPos > chromWindows {
```

Comment:
note to self: decrement here means to subtract previous searchSpace windows.

Example: the searchSpace consists of 3 regions in the genome, chr1 1 11, chr2 5 15, chr3 2 12. Say there are 6 windows in total in this searchSpace, 2 in each of the 3 regions, randPos is generated as an int between 0 and 6, and if randPos is 5, then it will look in region 1 (windows 1 and 2 are here), region 2 (windows 3 and 4 are here), and region 3 (windows 5 and 6 are here), land in region 3, 5-4=1 (line 40 randPos -= chromWindows), report window 5 out of all 6 windows